corehq/apps/ip_access/migrations/0001_initial.py

```diff
@@ -0,0 +1,56 @@
+# Generated by Django 4.2.16 on 2025-03-21 17:26
+
+import django.contrib.postgres.fields
+from django.db import migrations, models
+
+
+class Migration(migrations.Migration):
+
+    initial = True
+
+    dependencies = []
+
+    operations = [
+        migrations.CreateModel(
+            name="IPAccessConfig",
+            fields=[
+                (
+                    "id",
+                    models.AutoField(
+                        auto_created=True,
+                        primary_key=True,
+                        serialize=False,
+                        verbose_name="ID",
+                    ),
+                ),
+                ("domain", models.CharField(unique=True, db_index=True, max_length=126)),
+                (
+                    "country_allowlist",
+                    django.contrib.postgres.fields.ArrayField(
+                        base_field=models.CharField(max_length=2),
+                        default=list,
+                        size=None,
+                    ),
+                ),
+                (
+                    "ip_allowlist",
+                    django.contrib.postgres.fields.ArrayField(
+                        base_field=models.GenericIPAddressField(),
+                        default=list,
+                        size=None,
+                    ),
+                ),
+                (
+                    "ip_denylist",
```

Comment:
is there any need for a `country_denylist`?

Reply:
I left it out for now since we don't need it and it makes the logic a bit harder to reason about.  Though thinking about it again I think we could make it work by saying specificity wins (ie an IP rule beats a country rule), something like:

```python
def is_allowed(ip):
    if ip in ip_denylist:
        return False
    if ip in ip_allowlist:
        return True
    if country_denylist and is_in_country(ip, country_denylist):
        return False
    return not country_allowlist or is_in_country(ip, country_allowlist):
```

That is, allowlisted IPs are always allowed even from denylisted countries.

Still, I think it's okay to hold off for now - we could always add it later if needed